src/main/scala/org/constellation/snapshotstreaming/node/NodeDownload.scala

```diff
@@ -0,0 +1,56 @@
+package org.constellation.snapshotstreaming.node
+
+import cats.effect.Async
+
+import scala.concurrent.duration._
+
+import org.tessellation.dag.snapshot.GlobalSnapshot
+import org.tessellation.security.signature.Signed
+
+import fs2.Stream
+import org.constellation.snapshotstreaming.Configuration
+import org.constellation.snapshotstreaming.node.NodeClient
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait NodeDownload[F[_]] {
+  def downloadLatestOrdinal(): Stream[F, SnapshotOrdinal]
+  def downloadSnapshot(ordinal: SnapshotOrdinal): Stream[F, Either[SnapshotOrdinal, Signed[GlobalSnapshot]]]
+}
+
+object NodeDownload {
+
+  def make[F[_]: Async](nodeClient: NodeClient[F], config: Configuration) = new NodeDownload[F] {
+
+    private val logger = Slf4jLogger.getLoggerFromClass[F](NodeDownload.getClass)
+
+    override def downloadLatestOrdinal(): Stream[F, SnapshotOrdinal] = (for {
+      _ <- Stream.sleep(config.nodeIntervalInSeconds.seconds)
+      ordinal <- Stream
+        .eval(nodeClient.getLatestOrdinal)
+        .handleErrorWith(ex =>
+          Stream.eval(
+            logger.error(ex)(s"Couldn't download latest ordinal error from ${nodeClient.uri}.")
+          ) >> Stream.empty
+        )
+        .evalTap(ordinal => logger.debug(s"Latest ordinal from ${nodeClient.uri} is $ordinal."))
+    } yield ordinal).repeat
```

Comment:
I think that slow requests may slow down the whole stream. Instead of `Stream.sleep` I'd consider `awakeEvery` and `metered` eventually 🤔